packages/discord.js-utilities/src/lib/PromptMessage.ts

```diff
@@ -0,0 +1,150 @@
+import type {
+	Message,
+	User,
+	TextChannel,
+	NewsChannel,
+	APIMessageContentResolvable,
+	MessageOptions,
+	MessageAdditions,
+	DMChannel,
+	CollectorFilter,
+	EmojiIdentifierResolvable,
+	MessageReaction
+} from 'discord.js';
+
+/**
+ * This is a [[PromptMessage]], a utility to send a message with Yes/No promt.
+ * You must either use this class directly or extend it.
+ *
+ * [[PromptMessage]] uses actions, these are essentially reaction emojis, when triggered run the said action.
+ * You can utilize your own actions, or you can use the [[PromptMessage.defaultPrompts]].
+ * [[PromptMessage.defaultPrompts]] is also static so you can modify these directly.
+ *
+ * @example
+ * ```typescript
+ * const handler = new PromptMessage('Are you sure you want to continue?');
+ * const confirmed = handler.run();
+ * ```
+ *
+ * @example
+ * ```typescript
+ * const handler = new PromptMessage('Are you happy or sad?', {
+ * 		confirm: '�',
+ *		cancel: '�'
+ * });
+ * const confirmed = handler.run();
+ * ```
+ *
+ */
+export class PromptMessage {
+	/**
+	 * The message that has been sent in [[PromptMessage.run]]
+	 */
+	public message: IPromptMessage | null = null;
+
+	/**
+	 * The message that has been sent in [[PromptMessage.run]]
+	 */
+	public sentMessage: Message | null = null;
+
+	/**
+	 * The used options for this instance of [[PromptMessage]]
+	 */
+	public options: IPromptMessageOptions;
+
+	/**
+	 * Constructor for the [[PromptMessage]] class
+	 * @param message The message to send.
+	 * @param options Overrideable options if needed.
+	 */
+	public constructor(message: IPromptMessage, options?: Partial<IPromptMessageOptions>) {
+		this.message = message;
+		this.options = {
+			...PromptMessage.defaultOptions,
+			...(options ?? {})
+		};
+	}
+
+	/**
+	 * This executes the [[PromptMessage]] and sends the message.
+	 * The handler will start collecting 1 reaction and return a boolean corresponding the prompt.
+	 * @param channel The channel to use.
+	 * @param author The author to validate.
+	 */
+	public async run(channel: TextChannel | NewsChannel | DMChannel, author: User): Promise<IPromptMessageExplicitReturn | boolean> {
+		const { confirm, cancel, timeout, explicitReturn } = this.options;
+
+		if (!channel) throw new Error('There is no channel provided.');
+		if (!author) throw new Error('There is no author provided.');
+		if (!this.message) throw new Error('There is no messages.');
```

Comment:
```suggestion
		if (!channel) throw new SyntaxError('There is no channel provided.');
		if (!author) throw new SyntaxError('There is no author provided.');
		if (!this.message) throw new SyntaxError('There is no messages.');
```

extra clarity goes brrrrrr

Reply:
Copy pasta from `PaginatedMessage` Kek